Upstream-Status: Pending

Fix the segmentation fault issue when using dd on e500v2 targets,
use %f instead of $Lf, since single-float is used for e500v2 due to
an ICE of gcc-4.8.1.

Signed-off-by: Zhenhua Luo <zhenhua.luo@freescale.com>

--- coreutils-8.21/lib/human.c.orig	2013-08-16 05:17:24.466225177 -0500
+++ coreutils-8.21/lib/human.c	2013-08-16 05:17:52.007225123 -0500
@@ -228,7 +228,7 @@
 
     if (! (opts & human_autoscale))
       {
-        sprintf (buf, "%.0Lf", adjust_value (inexact_style, damt));
+        sprintf (buf, "%.0f", adjust_value (inexact_style, damt));
         buflen = strlen (buf);
         nonintegerlen = 0;
       }
@@ -246,7 +246,7 @@
 
         damt /= e;
 
-        sprintf (buf, "%.1Lf", adjust_value (inexact_style, damt));
+        sprintf (buf, "%.1f", adjust_value (inexact_style, damt));
         buflen = strlen (buf);
         nonintegerlen = decimal_pointlen + 1;
 
@@ -254,7 +254,7 @@
             || ((opts & human_suppress_point_zero)
                 && buf[buflen - 1] == '0'))
           {
-            sprintf (buf, "%.0Lf",
+            sprintf (buf, "%.0f",
                      adjust_value (inexact_style, damt * 10) / 10);
             buflen = strlen (buf);
             nonintegerlen = 0;
